src/main/java/com/redhat/cloud/custompolicies/app/rest/PolicyCrudService.java

```diff
@@ -338,8 +339,16 @@ public Response deletePolicy(@PathParam("id") UUID policyId) {
       builder = Response.status(Response.Status.BAD_REQUEST);
     } else {
       policy.delete(policy);
-      engine.deleteTrigger(policy.id, user.getAccount());
-      builder = Response.ok(policy);
+      try {
+        engine.deleteTrigger(policy.id, user.getAccount());
+        builder = Response.ok(policy);
+      } catch (NotFoundException nfe) {
+        // Engine does not have it - we can delete anyway
+        builder = Response.ok(policy);
+      } catch (Exception e) {
+        e.printStackTrace();  // TODO: Customise this generated block
+        builder = Response.serverError().entity(new Msg(e.getMessage()));
```

Comment:
which error code is it going to produce? is it server error 500?

Reply:
yes